src/labkey/constants.ts

```diff
@@ -14,40 +14,90 @@
  * limitations under the License.
  */
 export interface Container {
+    activeModules: Array<string>
+    folderType: string
+    formats: {
+        dateFormat: string
+        dateTimeFormat: string
+        numberFormat: string
+    }
+    hasRestrictedActiveModule: boolean
+    iconHref: string
+    id: string
+    isContainerTab: boolean
+    isWorkbook: boolean
+    name: string
+    parentId: string
+    parentPath: string
     path: string
+    sortOrder: number
+    startUrl: string
+    title: string
+    type: string
 }
 
-export interface ExperimentalFeatures {
-    containerRelativeURL: boolean
+export enum ExperimentalFeatures {
+    containerRelativeURL = 'containerRelativeURL',
+    disableGuestAccount = 'disableGuestAccount',
+    javascriptErrorServerLogging = 'javascriptErrorServerLogging',
+    javascriptMothership = 'javascriptMothership',
+    useExperimentalCoreUI = 'useExperimentalCoreUI',
+    strictReturnUrl = 'strictReturnUrl',
+}
+
+export type ExperimentalFlags = {
+    [key in ExperimentalFeatures]: boolean
 }
 
 export const CSRF_HEADER = 'X-LABKEY-CSRF';
 
-export interface LabKey {
-    container: Container
+export type LabKey = {
```

Comment:
I think we have several places in the various platform module main.d.ts files which are replicating this type definition. Since we are now exporting this from @labkey/api, it will be possible to make use of that in those places right?

Reply:
Definitely, and that's the idea. We no longer need to redefine these types in all of our apps. You can see an example of how this is used via [getServerContext](https://github.com/LabKey/labkey-ui-components/blob/72da98c289d8512e4f5412eb94bf78c437f4a8bb/packages/components/src/components/error/ErrorPage.tsx#L24) and via [global constant](https://github.com/LabKey/labkey-ui-components/blob/72da98c289d8512e4f5412eb94bf78c437f4a8bb/packages/components/src/typings/main.d.ts#L19).

Reply:
Thank god.